rich/spinner.py

```diff
@@ -0,0 +1,47 @@
+from time import sleep
+from threading import Thread
+
+RECALL = "\033[2K\r"
+
+
+class Spinner:
+    SLASHES = "|/-\\"
+
+    def loop(self):
+        i = 0
+        while self._run:
+            print(
+                "{r}{icon} {message}".format(
+                    r=RECALL,
+                    icon=self._icons[i % len(self._icons)],
+                    message=self.message,
+                ),
+                end="",
+            )
+            i += 1
+            sleep(self._delay)
+
+    def start(self):
+        self._run = True
```

Comment:
Better to use threading primitives as a way of communicating across threads. Have a look in the docs for the Event class.